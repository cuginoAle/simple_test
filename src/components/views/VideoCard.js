import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import ReactPlayer from 'react-player';
import Spinner from 'components/common/spinner';

const videoRatio = (9 * 100) / 16; // this is 16/9 ratio

const defaultVideoSettings = {
  controls: true,
  height: '100%',
  muted: true,
  playing: false,
  preload: 'auto',
  width: '100%',
};

const Wrapper = styled.div`
  border-radius: 5px;
  box-shadow: 0 1px 4px rgba(0, 0, 0, 0.6);
  overflow: hidden;
  padding-bottom: ${videoRatio}%;
  position: relative;
  background-color: rgba(255, 255, 255, 0.4);
  &:before {
    content: "";
    position: absolute;
    width: calc(100% + 5px);
    height: calc(100% + 5px);
    top: -5px;
    left: -5px;
    border: 5px solid transparent;
  }

  &:hover {
    &:before {
      border-color: red;
    }
  }
  > div {
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
  }

  > .placeholder {
    font-size: 3rem;
    color: rgba(0, 0, 0, 0.3);
    display: flex;
    align-items: center;
    justify-content: center;
  }
`;

const VideoCard = ({ className, videoDetails }) => {
  const classes = ['VideoCard'];
  const [videoIsReady, setVideoIsReady] = React.useState(false);
  const [videoSettings, setVideoSettings] = React.useState(
    defaultVideoSettings,
  );

  if (className) classes.push(className);

  function handleMouseEnter() {
    setVideoSettings({
      ...videoSettings,
      playing: true,
    });
  }

  function handleMouseLeave() {
    setVideoSettings({ ...videoSettings, playing: false });
  }

  function onReady() {
    setVideoIsReady(true);
  }

  return (
    <Wrapper
      className={classes.join(' ')}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      {videoDetails.src ? (
        <>
          {!videoIsReady && <Spinner relativeToParent />}
          <ReactPlayer
            url={videoDetails.src}
            {...videoSettings}
            onReady={onReady}
          />
        </>
      ) : (
        <div className="placeholder">
          Other video
          &nbsp;
          {videoDetails.key}
        </div>
      )}
    </Wrapper>
  );
};

VideoCard.displayName = 'VideoCard';

VideoCard.propTypes = {
  className: PropTypes.string,
  videoDetails: PropTypes.shape({
    key: PropTypes.string.isRequired,
    src: PropTypes.string,
  }).isRequired,
};

VideoCard.defaultProps = {
  className: null,
};

export default VideoCard;
