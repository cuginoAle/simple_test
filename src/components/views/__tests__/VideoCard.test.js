import { render } from '@testing-library/react';
import React from 'react';
import VideoCard from '../VideoCard';

describe('VideoCard', () => {
  it('Should match spanshot for video', () => {
    const videoDetails = {
      key: 'test',
      src: 'some url...',
    };
    const testComponent = render(<VideoCard videoDetails={videoDetails} />);

    expect(testComponent).toMatchSnapshot();
  });

  it('Should match spanshot for dummy card', () => {
    const videoDetails = {
      key: 'test',
    };
    const testComponent = render(<VideoCard videoDetails={videoDetails} />);

    expect(testComponent).toMatchSnapshot();
  });
});
