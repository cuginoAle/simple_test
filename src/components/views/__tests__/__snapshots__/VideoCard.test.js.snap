// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCard Should match spanshot for dummy card 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dlfnbm rLhzC VideoCard"
      >
        <div
          class="placeholder"
        >
          Other video  
          test
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dlfnbm rLhzC VideoCard"
    >
      <div
        class="placeholder"
      >
        Other video  
        test
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`VideoCard Should match spanshot for video 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dlfnbm rLhzC VideoCard"
      >
        <div
          class="sc-bdfBwQ cudsGs busySpinner"
        >
          <div
            class="sc-gsTCUz bsFaMp"
          />
        </div>
        <div
          preload="auto"
          style="width: 100%; height: 100%;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dlfnbm rLhzC VideoCard"
    >
      <div
        class="sc-bdfBwQ cudsGs busySpinner"
      >
        <div
          class="sc-gsTCUz bsFaMp"
        />
      </div>
      <div
        preload="auto"
        style="width: 100%; height: 100%;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
